tools/wptrunner/wptrunner/browsers/firefox.py

```diff
@@ -173,6 +192,8 @@ def __init__(self, logger, binary, prefs_root, test_type, extra_prefs=None, debu
         self.stylo_threads = stylo_threads
         self.chaos_mode_flags = chaos_mode_flags
 
+        self.server_locations = write_server_locations_file(self.config)
```

Comment:
I was just copying what we did with Servo for the hosts file, FWIW.

`tempfile.NamedTemporaryFile` doesn't work because there's no guarantee whether you can reopen the file a second time. Per the docs:

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).

I'd somehow missed the `cleanup` function before, bleh.